audio/out/ao_audiotrack.c

```diff
@@ -372,26 +372,26 @@ static uint32_t AudioTrack_getPlaybackHeadPosition(struct ao *ao)
         return 0;
     JNIEnv *env = MP_JNI_GET_ENV(ao);
     uint32_t pos = 0;
-    int64_t now = mp_raw_time_us();
+    int64_t now = mp_raw_time_ns();
     int state = MP_JNI_CALL_INT(p->audiotrack, AudioTrack.getPlayState);
 
     int stable_count = 20;
-    int64_t wait = p->timestamp_stable < stable_count ? 50000 : 3000000;
+    int64_t wait = p->timestamp_stable < stable_count ? 50000000 : 3000000000;
 
     if (state == AudioTrack.PLAYSTATE_PLAYING && p->format != AudioFormat.ENCODING_IEC61937 &&
         (p->timestamp_fetched == 0 || now - p->timestamp_fetched >= wait)) {
         if (!p->timestamp_fetched)
             p->timestamp_stable = 0;
 
-        int64_t utime1 = MP_JNI_GET_LONG(p->timestamp, AudioTimestamp.nanoTime) / 1000;
+        int64_t nanotime1 = MP_JNI_GET_LONG(p->timestamp, AudioTimestamp.nanoTime);
         if (MP_JNI_CALL_BOOL(p->audiotrack, AudioTrack.getTimestamp, p->timestamp)) {
             p->timestamp_set = true;
             p->timestamp_fetched = now;
             if (p->timestamp_stable < stable_count) {
                 uint32_t fpos = 0xFFFFFFFFL & MP_JNI_GET_LONG(p->timestamp, AudioTimestamp.framePosition);
-                int64_t utime2 = MP_JNI_GET_LONG(p->timestamp, AudioTimestamp.nanoTime) / 1000;
-                //MP_VERBOSE(ao, "getTimestamp: fpos= %u / time= %"PRId64" / now= %"PRId64" / stable= %d\n", fpos, utime2, now, p->timestamp_stable);
-                if (utime1 != utime2 && utime2 != 0 && fpos != 0) {
+                int64_t nanotime2 = MP_JNI_GET_LONG(p->timestamp, AudioTimestamp.nanoTime);
+                //MP_VERBOSE(ao, "getTimestamp: fpos= %u / time= %"PRId64" / now= %"PRId64" / stable= %d\n", fpos, nanotime2, now, p->timestamp_stable);
+                if (nanotime1 != nanotime2 && nanotime2 != 0 && fpos != 0) {
```

Comment:
what do these changes have to do with the timer code?

also a bit lower the calculations still work with microseconds

Edit: I noticed you removed `mp_raw_time_us`, I think this code can just use `mp_time_us`.

Reply:
We get ns timestamp, see `MP_JNI_GET_LONG(p->timestamp, AudioTimestamp.nanoTime)`, no need to keep truncating after replacing `mp_raw_time_us` with `mp_raw_time_ns`, it is jsut related change to that.

Reply:
Sure but this code just below still uses microseconds:
![grafik](https://github.com/mpv-player/mpv/assets/1042418/dc86455d-91e5-4707-b438-e52ce2e0771b)

I'd rather have as few as possible things changed in this file or all code checked and changed to use ns at once.
Since this isn't even in scope for the PR I prefer the former.